frontend/lib/providers/filtered_test_execution_ids.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:dartx/dartx.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import 'artefact_builds.dart';
+import 'test_execution_filters.dart';
+
+part 'filtered_test_execution_ids.g.dart';
+
+@riverpod
+Set<int> filteredTestExecutionIds(
```

Comment:
Just curious, does it make any difference that here we use `Set<int>` instead of `Set<TestExecution>` that way we wouldn't need to convert the IDs back to test executions in the widget code?

Reply:
Issue here is that to keep the same current UI we need to iterate over test executions per artefact build. So having a list or set of test executions wont help. Instead, we have to iterate over artefact builds and check each test execution inside it. I considered moving artefact builds into test executions and changing the UI to just show test executions. But I figured that's too much change to include in this PR

Reply:
Ah I see, I forgot about the existance of ArtefactBuilds in the UI, we can keep it as it is now.